### PR TITLE
Updated redirects

### PR DESCRIPTION
**Pull request by sachinSingh16-09**
#### References to other Issues or PRs
#24140 
#### Brief description of what is fixed or changed
Updated a bunch of permanent redirects

#### Release Notes

NO ENTRY

### CODE REVIEW

**Comment by haru-44** [comment]
I think the DOI link should not be modified.

https://github.com/sympy/sympy/pull/25505#discussion_r1293051942

**Comment by sachinSingh16-09** [comment]
Ohkk I will make the changes..

**Comment by haru-44** [comment]
* "https://help.github.com/en/articles/set-up-git" redirects to "https://docs.github.com/en/get-started/getting-started-with-git/set-up-git"
* Links to "docs.github.com" should use "https://docs.github.com/foobar" instead of "https://docs.github.com/en/foobar". Redirect to a page appropriate for the user's language environment.
* "https://mae.ufl.edu/~fregly/PDFs/autolev_tutorial.pdf" is probably a broken link.
* Why did you fix "http://www.eecis.udel.edu/~saunders/papers/sffge/it5.ps" ?

**Comment by sachinSingh16-09** [comment]
- "https://help.github.com/en/articles/set-up-git" redirects to "https://docs.github.com/en/get-started/getting-started-with-git/set-up-git" - it's already changed
Screenshot 2024-01-19 193650

- Links to "docs.github.com" should use "https://docs.github.com/foobar" instead of "https://docs.github.com/en/foobar". Redirect to a page appropriate for the user's language environment. - I have made the changes, Thanks.

- "https://mae.ufl.edu/~fregly/PDFs/autolev_tutorial.pdf" is probably a broken link. - I have already changed it
Screenshot 2024-01-19 194252

- Why did you fix "http://www.eecis.udel.edu/~saunders/papers/sffge/it5.ps" ? - This link is downloading a file which is insecure

Screenshot 2024-01-19 195157

so, I decided to change it, but I replaced with with a wrong link. I have changed it with the new one.

Screenshot 2024-01-19 195456

this works fine

Screenshot 2024-01-19 195811

File: doc/src/contributing/new-contributors-guide/dev-setup.md [hunk 1]
```diff
@@ -106,19 +106,19 @@ to do them again.
...
+you have not one yet then sign up at <https://github.com/join>. Your
 GitHub account is your presence in the open source world, so we recommend
 choosing a professional username.
 
 ### Setup SSH Keys
 
 To establish a secure connection between your computer and GitHub see detailed
-instructions in <https://help.github.com/en/articles/set-up-git> or at
-<https://help.github.com/en/articles/adding-a-new-ssh-key-to-your-github-account>.
+instructions in <https://docs.github.com/get-started/quickstart/set-up-git> or at
```
Comment (haru-44) [inline doc/src/contributing/new-contributors-guide/dev-setup.md:116]:
```suggestion
instructions in <https://docs.github.com/get-started/getting-started-with-git/set-up-git> or at
```

File: doc/src/modules/physics/mechanics/autolev_parser.rst [hunk 2]
```diff
@@ -564,7 +564,7 @@ Future Improvements
 1. Completing Dynamics Online
 -----------------------------
 The parser has been built by referring to and parsing codes from the
-`Autolev Tutorial <https://mae.ufl.edu/~fregly/PDFs/autolev_tutorial.pdf>`_
+`Autolev Tutorial <https://mae.ufl.edu/>`_
```
Comment (haru-44) [inline doc/src/modules/physics/mechanics/autolev_parser.rst:567]: Not a proper link fix.

File: sympy/physics/control/control_plots.py [hunk 3]
```diff
@@ -483,7 +483,7 @@ def impulse_response_plot(system, color='b', prec=8, lower_limit=0,
     References
     ==========
 
-    .. [1] https://www.mathworks.com/help/control/ref/lti.impulse.html
+    .. [1] https://www.mathworks.com/help/control/ref/dynamicsystem.impulse.html;jsessionid=30cd69fd0823a5e36e760497631b
```
Comment (haru-44) [inline sympy/physics/control/control_plots.py:486]:
```suggestion
    .. [1] https://www.mathworks.com/help/control/ref/dynamicsystem.impulse.html
```

File: doc/src/modules/physics/mechanics/autolev_parser.rst (hunk 2 above)
Comment (haru-44) [inline doc/src/modules/physics/mechanics/autolev_parser.rst:567]:
```suggestion
`Autolev Tutorial <https://mae.ufl.edu/~fregly/PDFs/autolev_tutorial.pdf>`_
```

If you are unsure of the appropriate link, you may leave it alone. Let's merge other fixes.

**Pull request merged**